src/lib.rs

```diff
@@ -1,6 +1,147 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
+extern crate memmap;
+extern crate byteorder;
+extern crate time;
+#[macro_use] extern crate bitflags;
+#[cfg(unix)] extern crate nix;
+#[cfg(windows)] extern crate kernel32;
+#[cfg(test)] extern crate rand;
+
+use byteorder::{LittleEndian, WriteBytesExt};
+use memmap::{Mmap, Protection};
+use std::ffi::CString;
+use std::fs::OpenOptions;
+use std::io::Cursor;
+use std::io::prelude::*;
+
+const HDR_LEN: usize = 40;
+const TOC_BLOCK_LEN: usize = 16;
+
+type Endian = LittleEndian;
+
+#[cfg(unix)]
+fn get_process_id() -> i32 {
+    nix::unistd::getpid()
+}
+
+#[cfg(windows)]
+fn get_process_id() -> i32 {
+    unsafe { kernel32::GetCurrentProcessId() as i32 }
+}
+ 
+bitflags! {
+    pub struct MMVFlags: u32 {
```

Comment:
Please add documentation for all public types and functions. Also, once published, we can add a https://docs.rs link to the readme for documentation instead of generating gh-pages ourselves, or we can do both. I think it is possible to automate that using travis. Just for every commit to master, it would regenerate doc and put it in the gh-pages branch.